invalid memory address or nil pointer dereference in formattedReviewerState

### Describe the bug

Attempt to view certain pull requests fails while attempting to print formattedReviewerState.

```
> gh --version
gh version 0.8.0 (2020-05-11)
```

### Steps to reproduce the behavior

Reproduced while running `gh pr view` for a particular PR.

Unfortunately I cannot share further information about the request. It is in a private repository.

I can say that the PR did have reviewers (4) and that none had approved.

### Expected vs actual behavior

Result is a runtime failure with the output provided in the next section.

### Logs

```
REDACTED_ISSUE_TITLE
Open • REDACTED_USER_NAME wants to merge 1 commit into master from REDACTED_BRANCH_NAME

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x173af80]

goroutine 1 [running]:
github.com/cli/cli/command.formattedReviewerState(0xc0002837a0, 0xc0005c8e80, 0x17)
	/home/runner/work/cli/cli/command/pr.go:505 +0xf0
github.com/cli/cli/command.prReviewerList(0xc0005c8e40, 0x20, 0x260f, 0xc000940f80, 0x3d, 0xc000357974, 0x4, 0x0, 0xc00061a480, 0x30, ...)
	/home/runner/work/cli/cli/command/pr.go:516 +0x14c
github.com/cli/cli/command.printPrPreview(0x1a8da80, 0xc00000e018, 0xc0004664e0, 0x7ffeefbffb23, 0x4)
	/home/runner/work/cli/cli/command/pr.go:436 +0x46b
github.com/cli/cli/command.prView(0x1fb1020, 0xc0003cd7c0, 0x1, 0x1, 0x0, 0x0)
	/home/runner/work/cli/cli/command/pr.go:346 +0x2fc
github.com/spf13/cobra.(*Command).execute(0x1fb1020, 0xc0003cd690, 0x1, 0x1, 0x1fb1020, 0xc0003cd690)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:840 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x1faee00, 0x5, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:945 +0x317
main.main()
	/home/runner/work/cli/cli/cmd/gh/main.go:32 +0xde
```

reproduction in the public repository

```bash
$ gh --version
gh version 0.8.0 (2020-05-11)
https://github.com/cli/cli/releases/tag/v0.8.0
```

```bash
$ gh pr view https://github.com/NG-ZORRO/ng-zorro-antd/pull/5311

docs: update docs and fix demo
Open • hsuanxyz wants to merge 1 commit into master from docs/list

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x173af80]

goroutine 1 [running]:
github.com/cli/cli/command.formattedReviewerState(0xc00063f240, 0x2, 0x2)
	/home/runner/work/cli/cli/command/pr.go:505 +0xf0
github.com/cli/cli/command.prReviewerList(0xc00022bec0, 0x20, 0x14bf, 0xc00022bee0, 0x1e, 0xc00065c01c, 0x4, 0x0, 0xc000023740, 0x33, ...)
	/home/runner/work/cli/cli/command/pr.go:516 +0x14c
github.com/cli/cli/command.printPrPreview(0x1a8da80, 0xc00000e018, 0xc00018b380, 0x7ffeefbffaaa, 0x4)
	/home/runner/work/cli/cli/command/pr.go:436 +0x46b
github.com/cli/cli/command.prView(0x1fb1020, 0xc00049caf0, 0x1, 0x1, 0x0, 0x0)
	/home/runner/work/cli/cli/command/pr.go:346 +0x2fc
github.com/spf13/cobra.(*Command).execute(0x1fb1020, 0xc00049ca10, 0x1, 0x1, 0x1fb1020, 0xc00049ca10)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:840 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x1faee00, 0x5, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:945 +0x317
main.main()
	/home/runner/work/cli/cli/cmd/gh/main.go:32 +0xde
```

--- a/command/pr.go
+++ b/command/pr.go
@@ -623,6 +623,8 @@ const (
 	approvedReviewState         = "APPROVED"
 	changesRequestedReviewState = "CHANGES_REQUESTED"
 	commentedReviewState        = "COMMENTED"
+	dismissedReviewState        = "DISMISSED"
+	pendingReviewState          = "PENDING"
 )
 
 type reviewerState struct {
@@ -648,8 +650,14 @@ func colorFuncForReviewerState(state string) func(string) string {
 
 // formattedReviewerState formats a reviewerState with state color
 func formattedReviewerState(reviewer *reviewerState) string {
-	stateColorFunc := colorFuncForReviewerState(reviewer.State)
-	return fmt.Sprintf("%s (%s)", reviewer.Name, stateColorFunc(strings.ReplaceAll(strings.Title(strings.ToLower(reviewer.State)), "_", " ")))
+	state := reviewer.State
+	if state == dismissedReviewState {
+		// Show "DISMISSED" review as "COMMENTED", since "dimissed" only makes
+		// sense when displayed in an events timeline but not in the final tally.
+		state = commentedReviewState
+	}
+	stateColorFunc := colorFuncForReviewerState(state)
+	return fmt.Sprintf("%s (%s)", reviewer.Name, stateColorFunc(strings.ReplaceAll(strings.Title(strings.ToLower(state)), "_", " ")))
 }
 
 // prReviewerList generates a reviewer list with their last state
@@ -705,6 +713,9 @@ func parseReviewers(pr api.PullRequest) []*reviewerState {
 	// Convert map to slice for ease of sort
 	result := make([]*reviewerState, 0, len(reviewerStates))
 	for _, reviewer := range reviewerStates {
+		if reviewer.State == pendingReviewState {
+			continue
+		}
 		result = append(result, reviewer)
 	}
 

--- a/command/pr_test.go
+++ b/command/pr_test.go
@@ -477,7 +477,7 @@ func TestPRView_Preview(t *testing.T) {
 			fixture:   "../test/fixtures/prViewPreviewWithReviewersByNumber.json",
 			expectedOutputs: []string{
 				`Blueberries are from a fork`,
-				`Reviewers: DEF \(Commented\), def \(Changes requested\), ghost \(Approved\), xyz \(Approved\), 123 \(Requested\), Team 1 \(Requested\), abc \(Requested\)\n`,
+				`Reviewers: DEF \(Commented\), def \(Changes requested\), ghost \(Approved\), hubot \(Commented\), xyz \(Approved\), 123 \(Requested\), Team 1 \(Requested\), abc \(Requested\)\n`,
 				`blueberries taste good`,
 				`View this pull request on GitHub: https://github.com/OWNER/REPO/pull/12\n`,
 			},

--- a/test/fixtures/prViewPreviewWithReviewersByNumber.json
+++ b/test/fixtures/prViewPreviewWithReviewersByNumber.json
@@ -70,6 +70,24 @@
                 "login": ""
               },
               "state": "APPROVED"
+            },
+            {
+              "author": {
+                "login": "hubot"
+              },
+              "state": "CHANGES_REQUESTED"
+            },
+            {
+              "author": {
+                "login": "hubot"
+              },
+              "state": "DISMISSED"
+            },
+            {
+              "author": {
+                "login": "monalisa"
+              },
+              "state": "PENDING"
             }
           ]
         },

